### ISSUE TITLE
[test-arch] reset_global_singletons autouse fixture pollutes call graph and couples 18K tests to 50+ modules

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#220]
## Discovery

mycelium v0.1.11 + AST call graph analysis of test coverage (2026-05-30).

## Problem

`tests/conftest.py` has a `reset_global_singletons()` function decorated
`@pytest.fixture(autouse=True)` that runs **before and after every single test**
(18,320 × 2 = 36,640 calls per full suite run).

The function imports and calls methods from ~15 different modules:

```python
from tree_sitter_analyzer.core.analysis_engine import UnifiedAnalysisEngine
from tree_sitter_analyzer.formatters.formatter_registry import FormatterRegistry
from tree_sitter_analyzer.mcp.utils.search_cache import clear_cache
# ... ~12 more
```

## Impact on call graph analysis

Because every test file imports `conftest.py` (pytest injects it automatically),
the call graph records every test as calling all 50+ methods that
`reset_global_singletons` touches. This creates massive false-positive coverage:

- `HealthHistory.append` → 183 test files (actual: ~5)
- `CacheService.get/set` → 225 test files (actual: ~10)
- `test_go_plugin.py` appears to call `PythonPlugin.analyze_file` (it doesn't)

This makes automated coverage analysis via call graphs impossible — the fixture
noise dominates the signal.

## Impact on test isolation

- If any import inside the function fails silently (all are wrapped in
  `try/except (ImportError, AttributeError): pass`), the singleton is NOT reset,
  but the test suite continues without warning → intermittent failures that are
  hard to diagnose.
- All 15 modules are now tightly coupled to the fixture — adding a new singleton
  anywhere means updating conftest.py or risking cross-test contamination.

## Proposed fix

Split `_reset_all_singletons()` into module-local fixtures that tests opt into:

```python
# conftest.py — minimal autouse
@pytest.fixture(autouse=True)
def reset_async_tasks():
    yield
    # only the cleanup that ALL tests need

# In test files that need specific resets:
@pytest.fixture
def fresh_analysis_engine():
    UnifiedAnalysisEngine._reset_instance()
    yield
    UnifiedAnalysisEngine._reset_instance()
```

Benefits:
- Call graph becomes accurate for coverage analysis
- Tests fail loudly if a reset is forgotten (explicit dependency)
- Faster tests (no 50-module import chain per test)

**Comment by aimasteracc** [comment]
## 部分解决（静态调用图污染）

将 `tests/conftest.py` 中 `_reset_all_singletons()` 的直接导入语句改为 `importlib.import_module()` 动态字符串导入。

**效果：** 静态调用图工具（mycelium、codegraph）不再从 conftest.py 生成到 50+ 生产模块的虚假边，消除了让 18K 测试看起来都调用了所有单例方法的假阳性覆盖率。

**测试行为不变：** autouse fixture 保留，所有单例重置逻辑保持一致。16,019 测试通过。

**未完成：** 将 autouse fixture 拆分为模块级 fixture 的完整重构（需要各目录 conftest.py + pytest.mark 改造）留给独立 PR，风险较高。关联 commit：`5bedeb5e`
